src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -806,10 +807,12 @@ private LookupElement createLookupItem(CompletionItem item) {
         Icon icon = iconProvider.getCompletionIcon(kind);
         LookupElementBuilder lookupElementBuilder;
 
-        if (insertText != null && !insertText.equals("")) {
-            lookupElementBuilder = LookupElementBuilder.create(insertText);
-        } else if (label != null && !label.equals("")) {
-            lookupElementBuilder = LookupElementBuilder.create(label);
+        if (!Strings.isNullOrEmpty(label)) {
+            lookupElementBuilder = LookupElementBuilder.create(label, "");
```

Comment:
Do we have any specific reason to pass "" as the lookup element string?

Reply:
Well i think for label scenario it might not, but for EditText scenario it does create the LookupItem with the lookup item instead of the lookup string, we want to lookup string to be presentation string since thats what is used for matching the items for text at caret.

In STS4 we have EditText as ${xxx.yyy} where the presentation string becomes xxx.yyyy. I couldn't find any other way to get this working correctly. This is something i found in the original work of this plugin ;)

Reply:
Oh I see :) I'll check again with ballerina LS with your latest commit and merge.

Reply:
@NipunaRanasinghe did you test the latest commit with BAL LS ?

Reply:
@gayanper Unfortunately this still breaks the Ballerina LS completions. :/ I debugged and observed that,
- We need to set the text which should be inserted, as the lookup string.
- `withLookupString()` method seems to be not working somehow. (might be a bug)

So I'll add a fix and push to your fork branch, which will solve your text edit issue hopefully.So could you please check and verify if your issue is fixed with my changes? :)